tests/frontend/specs/timeslider_follow.js

```diff
@@ -1,55 +1,69 @@
-    var timePerRev = 900
-      , revs = 10;
-    this.timeout(revs*timePerRev+10000);
-    for(var i=0; i < revs; i++) {
-      setTimeout(function() {
-        // enter 'a' in the first text element
-        inner$("div").last().sendkeys('a\n');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-      }, timePerRev*i);
+  it("content as it's added to timeslider", async function() {
+    // send 3 revisions
+    let revs = 6;
+    let message = 'a\n\n\n\n\n\n\n\n\n\n';
+    let newLines = message.split('\n').length
+    for (let i=0;i<revs;i++){
+      await helper.edit(message, newLines*i + 1);
     }
 
-    setTimeout(function() {
-      // go to timeslider
-      $('#iframe-container iframe').attr('src', $('#iframe-container iframe').attr('src')+'/timeslider#0');
+    await helper.gotoTimeslider(0);
+    await helper.waitForPromise(function(){return helper.contentWindow().location.hash === '#0'})
 
-      setTimeout(function() {
-        var timeslider$ = $('#iframe-container iframe')[0].contentWindow.$;
-        var $sliderBar = timeslider$('#ui-slider-bar');
+    let originalTop = helper.contentWindow().$('#innerdocbody').offset();
 
-        var latestContents = timeslider$('#innerdocbody').text();
+    // set to follow contents as it arrives
+    helper.contentWindow().$('#options-followContents').prop("checked", true);
+    helper.contentWindow().$('#playpause_button_icon').click();
 
-        // set to follow contents as it arrives
-        timeslider$('#options-followContents').prop("checked", true);
+    let newTop;
+    return helper.waitForPromise(function(){
+      newTop = helper.contentWindow().$('#innerdocbody').offset();
+      return newTop.top < originalTop.top;
+    })
+  });
 
-        var originalTop = timeslider$('#innerdocbody').offset();
-        timeslider$('#playpause_button_icon').click();
+  it("only to lines that exist in the current pad view #4389", async function(){
+    let rev0text = helper.textLines()[0];
+    let rev1text = "\xa0"; // timeslider will add &nbsp;
+    let rev2text = 'Test line'
 
-        setTimeout(function() {
-          //make sure the text has changed
-          var newTop = timeslider$('#innerdocbody').offset();
-          expect( originalTop ).not.to.eql( newTop );
-          done();
-        }, 1000);
+    helper.padInner$('body').empty();
+    await helper.waitForPromise(function(){
+      return helper.commits.length === 1; 
+    })
+    await helper.edit(rev2text)
 
-      }, 2000);
-    }, revs*timePerRev);
-  });
+    await helper.gotoTimeslider();
+
+    // set to follow contents as it arrives
+    helper.contentWindow().$('#options-followContents').prop("checked", true);
+
+    helper.contentWindow().$('#leftstep').click();
+    await helper.waitForPromise(function(){
+      return helper.timesliderTextLines()[0] === rev1text;
```

Comment:
How does this test whether the browser is following the edits? It looks like it's only checking that the contents match.

Reply:
The test is intended to catch a special bug:
at rev 0 delete all content
When you are in the timeslider at rev 1 (empty pad with a single new line) going to rev 0, the changeset will be something like: insert a lot of characters, including X new lines (depending on how many new lines are in the default text).

The current behaviour was to get no. of lines from the changeset and theffollow logic was: at rev 1, go to line no. X and apply the changeset to get rev 0. This obviously failed because there was no "line X" at rev 1 and the lines of revision 0 were never displayed. That's why I check the line contents.

But you are right, it's in the timeslider_follow suite and should include a test for following the edits. I extended the test case to cover that